### PR TITLE
create new assembly for federated identity

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: build/build.md [hunk 1]
```diff
@@ -29,8 +29,8 @@
   - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Web.MicrosoftGraphBeta')`
   - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\ProjectTemplates')`
   - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Web.TokenCache')`
-  - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Web.Certificate')
-  - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Certificate')`
+  - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Web.Certificate')`
+  - [template-pack-and-sign-nuget.yaml](template-pack-and-sign-nuget.yaml) `('$(Build.SourcesDirectory)\src\Microsoft.Identity.Web.IdentityFederation')`
```
Comment (jmprieur) [inline build/build.md:33]: Good catch!

File: src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        public AzureFederatedTokenProvider(string? federatedClientId)
+        {
+            _federatedClientId = federatedClientId;
+            ClientAssertionProvider = GetSignedAssertionFromFederatedTokenProvider;
+        }
+
+        private readonly string? _federatedClientId;
+
+        /// <summary>
+        /// Prototype of certificate-less authentication using a signed assertion
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs:30]:
```suggestion
        /// Certificate-less authentication using a signed assertion
```

File: src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        /// acquired with MSI (federated identity).
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs:31]:
```suggestion
        /// acquired with Managed Identity (federated identity).
```

File: src/Microsoft.Identity.Web.IdentityFederation/Microsoft.Identity.Web.IdentityFederation.csproj [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net462'">
+    <DefineConstants>$(DefineConstants);DOTNET_462</DefineConstants>
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net472'">
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.IdentityFederation/Microsoft.Identity.Web.IdentityFederation.csproj:73]: not needed

File: src/Microsoft.Identity.Web.IdentityFederation/Microsoft.Identity.Web.IdentityFederation.csproj [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      <PackagePath></PackagePath>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup Label="Build Tools" Condition="$([MSBuild]::IsOsPlatform('Windows'))">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp3.1; net462; net472; net5.0; netstandard2.0</TargetFrameworks>
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.IdentityFederation/Microsoft.Identity.Web.IdentityFederation.csproj:54]: netstandard2.0 should be enough

**Review by jmprieur — APPROVED** [review]
Awesome!
Thanks @jennyf19

File: src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web.IdentityFederation/AzureFederatedTokenProvider.cs:30]: I tried to commit your suggestion directly but it wouldn't take. oh well. thanks again.

**Pull request merged**